hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/HoodieReadClient.java

```diff
@@ -92,6 +92,18 @@ public HoodieReadClient(HoodieSparkEngineContext context, String basePath, SQLCo
     this.sqlContextOpt = Option.of(sqlContext);
   }
 
+  /**
+   * @param context
+   * @param basePath
+   * @param sqlContext
+   * @param indexType
+   */
+  public HoodieReadClient(HoodieSparkEngineContext context, String basePath, SQLContext sqlContext, HoodieIndex.IndexType indexType) {
```

Comment:
Is this going to be used in any query engine?

Reply:
We use this client in our custom code. AFAIK this is the only way to benefit from bloom indexes while reading a hudi table. Then both global  bloom / bloom are useful